frontend/src/modules/WellCompletion/settings.tsx

```diff
@@ -185,60 +175,83 @@ export const settings = ({ moduleContext, workbenchSession, workbenchServices }:
     function handleRealizationSelectionChange(_: React.ChangeEvent<HTMLInputElement>, value: string | number) {
         const selection = value as RealizationSelection.Aggregated | RealizationSelection.Single;
         setRealizationSelection(selection);
-        if (selection === RealizationSelection.Aggregated) {
-            setSelectedRealizationNumber(undefined);
-        }
-        if (selection === RealizationSelection.Single && computedEnsemble?.getRealizations()) {
-            const realization = computedEnsemble.getRealizations()?.[0];
-            setSelectedRealizationNumber(realization);
-        }
     }
 
     function handleSelectedRealizationNumberChange(realizationNumber: string) {
         setSelectedRealizationNumber(parseInt(realizationNumber));
     }
 
     function handleTimeAggregationChange(_: React.ChangeEvent<HTMLInputElement>, value: string | number) {
-        const newTimeAggregation = value as TimeAggregation;
+        const newTimeAggregation = value as TimeAggregationType;
+        let newTimeStepIndex: number | [number, number] | null = 0;
         if (!availableTimeSteps) {
-            setSelectedTimeStepOptions({ timeStepIndex: 0, timeAggregation: newTimeAggregation });
+            setSelectedTimeStepOptions({ timeStepIndex: newTimeStepIndex, timeAggregationType: newTimeAggregation });
             return;
         }
 
-        if (newTimeAggregation === "Max" || newTimeAggregation === "Average") {
-            setSelectedTimeStepOptions((prev) => ({
-                timeStepIndex:
-                    typeof prev.timeStepIndex === "number" ? [0, availableTimeSteps.length - 1] : prev.timeStepIndex,
-                timeAggregation: newTimeAggregation,
-            }));
-            return;
+        if (newTimeAggregation === TimeAggregationType.Max || newTimeAggregation === TimeAggregationType.Average) {
+            newTimeStepIndex =
+                typeof selectedTimeStepOptions.timeStepIndex === "number"
+                    ? [0, availableTimeSteps.length - 1]
+                    : selectedTimeStepOptions.timeStepIndex;
+            setSelectedTimeStepOptions({
+                timeStepIndex: newTimeStepIndex,
+                timeAggregationType: newTimeAggregation,
+            });
+        } else if (newTimeAggregation === TimeAggregationType.None) {
+            newTimeStepIndex =
+                typeof selectedTimeStepOptions.timeStepIndex === "number"
+                    ? selectedTimeStepOptions.timeStepIndex
+                    : selectedTimeStepOptions.timeStepIndex
```

Comment:
I find it better readable if you'd split up this very long inline if statement and use regular if statements. In addition, here I also find it better readable if you'd explicitly test for `selectedTimeStepOptions.timeStepIndex !== null`.